x-pack/plugins/session_view/public/components/Process/index.tsx

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState, useEffect, MouseEvent } from 'react';
+import { EuiButton, EuiIcon, useEuiTheme } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { IProcess } from '../../hooks/use_process_tree';
+
+const TREE_INDENT = 32;
+
+interface IProcessDeps {
+  process: IProcess;
+  isSessionLeader?: boolean;
+  depth?: number;
+  onProcessSelected(process: IProcess): void;
+}
+
+/**
+ * Renders a node on the process tree
+ * TODO: as well as sections for tty output, alerts and file redirection.
+ */
+export function Process({
+  process,
+  isSessionLeader = false,
+  depth = 0,
+  onProcessSelected,
+}: IProcessDeps) {
+  const { euiTheme } = useEuiTheme();
+  const [childrenExpanded, setChildrenExpanded] = useState(isSessionLeader || process.autoExpand);
+
+  useEffect(() => {
+    setChildrenExpanded(isSessionLeader || process.autoExpand);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [process.autoExpand]);
+
+  const darkTextCSS = `
+    color: ${euiTheme.colors.text};
+  `;
+
+  const searchHighlightCSS = `
+    background-color: ${euiTheme.colors.highlight};
+    color: ${euiTheme.colors.text};
+    border-radius: 4px;
+  `;
+
+  const event = process.getLatest();
```

Comment:
curious to know what `getLatest` is for? seems a bit odd that theres a getter on a `process` which seems like it's just a JSON. Would it be better if we have `event` come in as a prop instead? 

React definitely recommends working with data directly, as opposed to classes/wrappers

Reply:
So, when these events flow in, there can be multiple events per process. e.g 'fork', 'exec' and 'exit' event, as well as possibly 'output' and 'alert' events. Each "Process" class houses the events for a particular process. and so when we access details of the event, we grab the most recent fork,exec, or exit and use that event when rendering. e.g if there is an exit event, we want info about exit_code which would be on the latest event. I'm not a huge fan of classes too, but due to the complex nature of this data and how it needs to rebuild hierarchy, it helps keep things clean. The alternative is a bunch of util functions that we need to pass around in places that need it.